### DISCUSSION TITLE
Renovate automerging 2 branches from 4 on one run instead of only 1 per run as stated in documentation

### DISCUSSION THREAD

**Discussion by vdobos-trask** [renovatebot/renovate#26372] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.102.3

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### What is your question?

I'm tuning renovatebot configuration for deployment on a larger java legacy product suite spread across few dozen git repositories, with main purpose being constant update of inter-project utility libraries.
Most of the repositories have only 1 active branch but we have several artifacts that are actively maintained across several distinct versions.
For example, I have artifact with 4 active versions (branches). All 4 branches are configured as baseBranches in renovate.json on mentioned project.
When I run renovate, it correctly recognizes all 4 branches, correctly updates pom.xml, but then automerges only 2 branches, leaving other 2 for second run. 
This would be fine, but from documentation, it is clear that renovate should perform only one automerge.

From "Known Limitations" page:
![image](https://github.com/renovatebot/renovate/assets/67316052/68e75ba9-aa39-4fad-8400-fea5b169ac5d)

In cli log I can find these lines (first after first merge, second line after second merge, few thousand lines lower):

INFO: Restarting repository job after automerge result (repository=...)

DEBUG: Automerged but already retried once (repository=...)

My question is:

**Is this a correct behaviour? If so, what is a reason for it ? Or is it a bug or unexpected side-effect of other functionality ?** 

I need to know this so we can properly schedule number of renovate executions to perform complete update of possible internal library releases every day.

More details:

I couldn't find any option to properly tune number of automerges. None of "prHourlyLimit", "branchConcurrentLimit", "prConcurrentLimit" produced desired behaviour, with "prHourlyLimit" being close but having problem of being "hourly".
 We set-up renovatebot to autoupdate its container images and this question is aimed on whether i should keep an eye on possible changes (that beeing renovate will suddenly start automerging all branches on one run or reduce number of automerged branches to one per run)

Thanks

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment]
Renovate will essentially retry/restart a job once if the first time an automerge happened. It won't repeat that if the second run also automerges.

I have a pushed an update to the documentation to fix/clarify this

**Comment by vdobos-trask** [comment]
Thanks for clarification
